tests/test_bookmarks.py

```diff
@@ -0,0 +1,125 @@
+                second_sync_count = second_sync_record_count.get(stream, 0)
+                first_sync_messages = [record.get('data') for record in first_sync_records.get(stream).get('messages')
+                                       if record.get('action') == 'upsert']
+                second_sync_messages = [record.get('data') for record in second_sync_records.get(stream).get('messages')
+                                        if record.get('action') == 'upsert']
+                first_bookmark_key_value = first_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_key_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+
+                if self.is_incremental(stream):
+
+                    # collect information specific to incremental streams from syncs 1 & 2
+                    replication_key = next(iter(self.expected_replication_keys()[stream]))
+                    first_bookmark_value = first_bookmark_key_value.get(replication_key)
+                    second_bookmark_value = second_bookmark_key_value.get(replication_key)
+                    first_bookmark_value_parsed = parse(first_bookmark_value).strftime("%Y-%m-%dT%H:%M:%SZ")
+                    second_bookmark_value_parsed = parse(second_bookmark_value).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+                    # Verify the first sync sets a bookmark of the expected form
+                    self.assertIsNotNone(first_bookmark_key_value)
+                    self.assertIsNotNone(first_bookmark_key_value.get(replication_key))
+
+                    # Verify the second sync sets a bookmark of the expected form
+                    self.assertIsNotNone(second_bookmark_key_value)
+                    self.assertIsNotNone(second_bookmark_key_value.get(replication_key))
+
+                    # Verify the second sync bookmark is Equal to the first sync bookmark
+                    self.assertEqual(second_bookmark_value, first_bookmark_value) # assumes no changes to data during test
+
+                    for record in second_sync_messages:
+
+                        # Verify the second sync bookmark value is the max replication key value for a given stream
+                        replication_key_value = record.get(replication_key)
+                        replication_key_value_parsed = parse(replication_key_value).strftime("%Y-%m-%dT%H:%M:%SZ")
+                        self.assertLessEqual(
+                            replication_key_value_parsed, second_bookmark_value_parsed,
+                            msg="Second sync bookmark was set incorrectly, a record with a greater replication-key value was synced."
+                        )
+
+                        # Verify the data of the second sync is greater-equal to the bookmark from the first sync
+                        self.assertGreaterEqual(
+                            replication_key_value_parsed, parse(self.second_start_date).strftime("%Y-%m-%dT%H:%M:%SZ"),
```

Comment:
Again can we change the variable name here.

Reply:
As per the above comment, added a comment in the code.